src/Take.Blip.Builder/FlowManager.cs

```diff
@@ -544,5 +546,25 @@ await output.Conditions.EvaluateConditionsAsync(lazyInput, context, cancellation
 
             return state;
         }
+
+        private string DefineMutexKey(Message message, Flow flow, Identity userIdentity)
+        {
+            string parent = flow.Id;
+            string identity = userIdentity.ToString();
+
+            if (message.Metadata != null && message.Metadata.ContainsKey(TUNNEL_OWNER_METADATA))
+            {
+                //when the message is traveling in the context of a router, we must consider the key as the router identifier and the tunnel originator
+                message.Metadata.TryGetValue(TUNNEL_OWNER_METADATA, out parent);
+                message.Metadata.TryGetValue(TUNNEL_ORIGINATOR_METADATA, out identity);
```

Comment:
Is it possible for it to have `TUNNEL_OWNER_METADATA` but not `TUNNEL_ORIGINATOR_METADATA`.

Also, `.TryGetValue` already returns a `bool` indicating if the value exists or not. You can use it on the `if` to avoid repetition.